app/scripts/metamask-controller.test.js

```diff
@@ -4544,17 +4549,20 @@ describe('MetaMaskController', () => {
         { entropySource: primaryId },
       );
 
-      expect(wallet.discoverAndCreateAccounts).toHaveBeenCalledTimes(1);
-      expect(result).toStrictEqual({ Bitcoin: 1, Solana: 2 });
+      expect(wallet.discoverAccounts).toHaveBeenCalledTimes(1);
+      expect(result).toStrictEqual({ Bitcoin: 0, Solana: 2 });
     });
 
     it('passes provided keyring id to wallet getter', async () => {
       const providedId = 'test-keyring-id';
 
       const wallet = {
-        discoverAndCreateAccounts: jest
+        discoverAccounts: jest
           .fn()
-          .mockResolvedValue({ Bitcoin: 1, Solana: 2 }),
+          .mockResolvedValue([
+            { type: SolAccountType.DataAccount },
+            { type: SolAccountType.DataAccount },
```

Comment:
We could do the same here too?

```suggestion
            { type: SolAccountType.DataAccount },
            { type: EthAccountType.Eoa },
```